From dd1eabf6ba0b2a42b034f0918bf0370307983709 Mon Sep 17 00:00:00 2001
From: Bandan Das <bsd@redhat.com>
Date: Wed, 7 Aug 2013 17:28:00 +0200
Subject: pc_piix: disable CPUID_SEP for 6.4.0 machine types and below

RH-Author: Bandan Das <bsd@redhat.com>
Message-id: <jpghaf1v2an.fsf@redhat.com>
Patchwork-id: 53052
O-Subject: [PATCH RHEL-7 qemu-kvm v2] pc_piix: disable CPUID_SEP for 6.4.0 machine types and below
Bugzilla: 960216
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Eduardo Habkost <ehabkost@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>

Bugzilla : 960216
Brew : https://brewweb.devel.redhat.com/taskinfo?taskID=6140166

With c88f862d9 the SEP flag was disabled for RHEL 6.4 and below
machine types due to a kernel bug.  3af51f9^..8f38a22 enabled it
back again for RHEL 6.5 machine type. This change implements
this rhel 6 compatibility behavior

Implementation differences : Simply call x86_cpu_compat_set_features,
NULL signifies all applicable CPU models

v2 : specify correct comment range for rhel 6 commits

Signed-off-by: Bandan Das <bsd@redhat.com>

diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index bc67633..9e03e52 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -881,6 +881,7 @@ static QEMUMachine pc_machine_rhel650 = {
 static void pc_init_rhel640(QEMUMachineInitArgs *args)
 {
     has_pvpanic = false;
+    x86_cpu_compat_set_features(NULL, FEAT_1_EDX, 0, CPUID_SEP);
     pc_init_rhel650(args);
 }
 
